master/internal/rm/kubernetesrm/kubernetes_resource_manager_intg_test.go

```diff
@@ -0,0 +1,176 @@
+//go:build integration
+// +build integration
+
+package kubernetesrm
+
+import (
+	"strconv"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	k8sV1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/determined-ai/determined/master/internal/mocks"
+	"github.com/determined-ai/determined/master/pkg/device"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/proto/pkg/apiv1"
+)
+
+const (
+	defaultNamespace   = "default"
+	auxNodeName        = "aux"
+	compNodeName       = "comp"
+	pod1NumSlots       = 4
+	pod2NumSlots       = 8
+	nodeNumSlots       = 8
+	nodeNumSlotsCPU    = 20
+	podServSlotTypeGPU = "randomDefault"
+)
+
+// Number of active slots on given nodes.
+var nodeToSlots = map[string]int{
+	auxNodeName:  pod1NumSlots,
+	compNodeName: pod2NumSlots,
+}
+
+func TestGetAgentsAndGetSlots(t *testing.T) {
+	// Create agent nodes.
+	auxNode := createMockNode(auxNodeName)
+	compNode := createMockNode(compNodeName)
+
+	nodes := make(map[string]*k8sV1.Node)
+	nodes[auxNode.Name] = &auxNode
+	nodes[compNode.Name] = &compNode
+
+	// Create resource managers with different device types for their respective pod service.
+	rmGPU := createMockRM(nodes, podServSlotTypeGPU)
+	rmCUDA := createMockRM(nodes, device.CUDA)
+	rmCPU := createMockRM(nodes, device.CPU)
+
+	rmList := []*ResourceManager{rmGPU, rmCUDA, rmCPU}
+	for _, rm := range rmList {
```

Comment:
use table driven tests for this; it is exactly what it is for. (see https://hpe-aiatscale.atlassian.net/wiki/spaces/ENGINEERIN/pages/1389035562/Writing+Tests+for+Backend+Code#Resources)